Hades: implement password reset link

akhilnarang opened the pull request:
Still needs to be tested and documented, and linked on the login page

akhilnarang committed:
Hades: implement password reset link

Still needs to be tested and documented, and linked on the login page

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

prkprime commented on hades/templates/reset_password.html:
```diff
@@ -0,0 +1,25 @@
+    <input type="password" id="inputNewPassword" class="form-control" placeholder="New password" name="new_password"
+           required>
+    <!--
+    TODO: Add JS validation to ensure password has been entered again
```
Do you mean to check if old password is used or to check if another password has been inputted on the same page?

akhilnarang replied:
I meant just to verify password, most change password pages make enter it twice for confirmation

alr0cks reviewed (commented):
It works!

prkprime reviewed (approved):
Everything seems fine. Also if anything breaks, #BlameAkeel xd

akhilnarang committed:
Hades: fix reset password route

Had to use `string` not `str` here

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: a few updates

Add a forgot username page as well, which requires your email
Ask just for username for forgot password, no need of email for that

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: templates: forgot_{username,password}: cleanup and fix

Fix allowed methods for `/forgot_username`

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: utils: we can iterate over `[]`, not `None`

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: `/reset_password`: fix link expiry check

  * Also display a message if changing password is successful, else you'll just be sent back to the same page
  * Also change type from password to text, since a username is definitely not a password (right?)

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang commented:
All functionality seems fine, UI could probably be improved...

akhilnarang committed:
Hades: update README with new environment variable required

Signed-off-by: Akhil Narang <me@akhilnarang.dev>
